### PR TITLE
Add documentation for `weather.get_forecasts`

### LINKED ISSUE #30184: No documentation for new get_forecasts service
### Feedback

Got a `repair` notification in home assistant to migrate to the new `get_forecasts` service call but no documentation for the new service exists.

### URL

https://www.home-assistant.io/integrations/weather

### Version

2023.12.0

### Additional information

https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/weather.markdown?plain=1#L73-L92

Seems to not be updated.

### PR DESCRIPTION
**Pull request by mikeodr**
## Proposed change

Add missing documentation for https://github.com/home-assistant/core/pull/102534

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes https://github.com/home-assistant/home-assistant.io/issues/30184

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by mikeodr** [comment]
Targeting `current` branch since this missed getting into the 2023.12 release notes.

If that's not correct let me know, can re-target to `next`

**Comment by joostlek** [comment]
Ignore the bot yes

File: source/_integrations/weather.markdown [hunk 1]
```diff
@@ -151,3 +158,104 @@ forecast:
     precipitation: 0
 ```
 {% endraw %}
+
+## Service `weather.get_forecasts`
+
+This service populates [Response Data](/docs/scripts/service-calls#use-templates-to-handle-response-data)
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
This service populates [response data](/docs/scripts/service-calls#use-templates-to-handle-response-data)
```

File: source/_integrations/weather.markdown [hunk 2]
```diff
@@ -151,3 +158,104 @@ forecast:
...
+with a list of weather services and their associated forecasts.
+
+| Service data attribute | Optional | Description | Example |
+| ---------------------- | -------- | ----------- | --------|
+| `type` | no | The type of forecast, must be one of `daily`, `twice_daily` or `hourly`. | daily
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
| `type` | no | The type of forecast, must be one of `daily`, `twice_daily`, or `hourly`. | daily
```

File: source/_integrations/weather.markdown [hunk 3]
```diff
@@ -151,3 +158,104 @@ forecast:
...
+  entity_id:
+    - weather.tomorrow_io_home_nowcast
+    - weather.toronto_forecast
+data:
+  type: daily
+response_variable: weather_forecast
+```
+{% endraw %}
+
+The response data field is a list of called target entities each containing the `forecast` field .
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
The response data field is a list of called target entities, each containing the `forecast` field.
```

File: source/_integrations/weather.markdown [hunk 4]
```diff
@@ -151,3 +158,104 @@ forecast:
...
+| `is_daytime` | Only set for `twice_daily` forecasts. | False
+| `apparent_temperature` | The apparent (feels-like) temperature in the unit indicated by the `temperature_unit` state attribute. | 10.2
+| `cloud_coverage` | The cloud coverage in %. | 15
+| `condition` | The weather condition. | Sunny
+| `dew_point` | The dew point temperature in the unit indicated by the `temperature_unit` state attribute. | 6.0
+| `humidity` | The relative humidity in %. | 82
+| `precipitation_probability` | The probability of precipitation in %. | 0
+| `precipitation` | The precipitation amount in the unit indicated by the `precipitation_unit` state attribute. | 0
+| `pressure` | The air pressure in the unit indicated by the `pressure_unit` state attribute. | 1019
+| `temperature` | The temperature in the unit indicated by the `temperature_unit` state attribute. If templow is also provided, this is the higher temperature. | 14.2
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
| `temperature` | The temperature in the unit indicated by the `temperature_unit` state attribute. If `templow` is also provided, this is the higher temperature. | 14.2
```

**Review by c0ffeeca7 — DISMISSED** [review]
LGTM, thank you for picking this up, @mikeodr :+1:

File: source/_integrations/weather.markdown [hunk 5]
```diff
@@ -151,3 +158,104 @@ forecast:
     precipitation: 0
 ```
 {% endraw %}
+
+## Service `weather.get_forecasts`
```
Comment (MartinHjelmare) [inline source/_integrations/weather.markdown:71]: I'd put this at the top and the deprecated service at the bottom. Alternatively remove the deprecated service altogether.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: source/_integrations/weather.markdown [hunk 6]
```diff
@@ -151,3 +158,104 @@ forecast:
...
+
+This service populates [response data](/docs/scripts/service-calls#use-templates-to-handle-response-data)
+with a list of weather services and their associated forecasts.
+
+| Service data attribute | Optional | Description | Example |
+| ---------------------- | -------- | ----------- | --------|
+| `type` | no | The type of forecast, must be one of `daily`, `twice_daily`, or `hourly`. | daily
+
+{% raw %}
```
Comment (klaasnicolaas) [inline source/_integrations/weather.markdown:120]: raw tags are only needed when there is jinja code used in the example

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_integrations/weather.markdown (hunk 5 above)
Comment (klaasnicolaas) [reply source/_integrations/weather.markdown:71]: I think we only remove it when it no longer exists, so as long users can still use it (or deprecated), it should still be there.

Comment (MartinHjelmare) [reply source/_integrations/weather.markdown:71]: Normally we remove deprecated features from the docs when we deprecate them. Sometimes we keep it around if it's a major feature. I can see us doing one or the other here.

File: source/_integrations/weather.markdown (hunk 6 above)
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown:120]:
```suggestion
```

File: source/_integrations/weather.markdown [hunk 7]
```diff
@@ -151,3 +158,104 @@ forecast:
...
+service: weather.get_forecasts
+target:
+  entity_id:
+    - weather.tomorrow_io_home_nowcast
+    - weather.toronto_forecast
+data:
+  type: daily
+response_variable: weather_forecast
+```
+{% endraw %}
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown:142]:
```suggestion
```

File: source/_integrations/weather.markdown [hunk 8]
```diff
@@ -151,3 +158,102 @@ forecast:
...
+| `templow` | The lower temperature in the unit indicated by the `temperature_unit` state attribute. | 5.0
+| `uv_index` | The UV index. | 3
+| `wind_bearing` | The wind bearing in azimuth angle (degrees) or 1-3 letter cardinal direction. | 268
+| `wind_gust_speed` | The wind gust speed in the unit indicated by the `wind_speed_unit` state attribute. | 34.41
+| `wind_speed` | The wind speed in the unit indicated by the `wind_speed_unit` state attribute. | 24.41
+
+
+Example service response:
+
+{% raw %}
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
```

File: source/_integrations/weather.markdown [hunk 9]
```diff
@@ -151,3 +158,102 @@ forecast:
...
+      precipitation_probability: 40
+      temperature: 3
+      templow: -2
+    - datetime: "2023-12-08T20:08:32.823861-05:00"
+      condition: cloudy
+      precipitation_probability: 0
+      temperature: 6
+      templow: 4
+```
+{% endraw %}
```
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown]:
```suggestion
```

File: source/_integrations/weather.markdown [hunk 10]
```diff
@@ -74,6 +74,13 @@ wind_speed_unit: km/h
 
 Weather integrations which support weather forecasts expose the forecast using services. The services provided by weather entities are described below, and you can also read more about [Service Calls](/docs/scripts/service-calls/).
 
+<div class="note warning">
+
+`weather.get_forecast` is deprecated and will be removed in a future release of Home Assistant Core.
+[`weather.get_forecasts`](/integrations/weather#service-weatherget_forecasts) should be used.
+
+</div>
```
Comment (frenck) [inline source/_integrations/weather.markdown]: This is not a changelog. We need to remove deprecated stuff.

File: source/_integrations/weather.markdown [hunk 11]
```diff
@@ -151,3 +158,102 @@ forecast:
...
+| `uv_index` | The UV index. | 3
+| `wind_bearing` | The wind bearing in azimuth angle (degrees) or 1-3 letter cardinal direction. | 268
+| `wind_gust_speed` | The wind gust speed in the unit indicated by the `wind_speed_unit` state attribute. | 34.41
+| `wind_speed` | The wind speed in the unit indicated by the `wind_speed_unit` state attribute. | 24.41
+
+
+Example service response:
+
+{% raw %}
+```yaml
```
Comment (frenck) [inline source/_integrations/weather.markdown]: Wrap this in a details block?

File: source/_integrations/weather.markdown (hunk 10 above)
Comment (c0ffeeca7) [reply source/_integrations/weather.markdown]: OK, will do. Thank you!

File: source/_integrations/weather.markdown [hunk 12]
```diff
@@ -151,3 +158,100 @@ forecast:
...
+| `type` | no | The type of forecast, must be one of `daily`, `twice_daily`, or `hourly`. | daily
+
+```yaml
+service: weather.get_forecasts
+target:
+  entity_id:
+    - weather.tomorrow_io_home_nowcast
+    - weather.toronto_forecast
+data:
+  type: daily
```
Comment (frenck) [inline source/_integrations/weather.markdown]: That is the default already

```suggestion
```

File: source/_integrations/weather.markdown (hunk 10 above)
Comment (c0ffeeca7) [inline source/_integrations/weather.markdown:82]:
```suggestion
```

**Comment by mikeodr** [comment]
Thanks all I'll update shortly.

Just wanted to get something rolling since I went to go find the docs and it was missing.

File: source/_integrations/weather.markdown (hunk 12 above)
Comment (mikeodr) [reply source/_integrations/weather.markdown]: I'll provide another example of non-default. Examples should show as many fields in use as possible. Gives users additional detail etc.

**Comment by mikeodr** [comment]
Updated, removed the deprecated per frenck's comment (thanks for the clarification). Set the example service call to use the non-default. Also specified that daily is the default. Wrapped the response example in a details block.

**Comment by eifinger** [comment]
I accidentally opened a duplicate to this. But I think in this PR we are missing an update on the template instructions:
https://github.com/home-assistant/home-assistant.io/pull/30204/files#diff-6dd9f7067162094a8a1436d32413cce0c014821daab4d12d23274e9035583e99R782-R793

**Comment by mikeodr** [comment]
I'll add that in a bit when I can, on the day job now, unless someone else on core beats me to it, that is a good example to include.

**Comment by mikeodr** [comment]
I added that update, had a spare cycle sooner than I thought. Thanks @eifinger

**Comment by mikeodr** [comment]
It occurred to me since I missed the item eifinger highlighted that I should grep for more instances. I found a python script example that needed updating also.

File: source/_integrations/weather.markdown [hunk 13]
```diff
@@ -68,37 +67,29 @@ wind_gust_speed: 51.56
...
-<div class='note'>
-
-Some integrations may not support the `weather.get_forecast` service, but instead expose weather forecasts as a state attribute named `forecast`. This behavior is deprecated and will be removed in a future release of Home Assistant Core.
-
-</div>
-
-This service populates [Response Data](/docs/scripts/service-calls#use-templates-to-handle-response-data)
-with a weather forecast.
+This service populates [response data](/docs/scripts/service-calls#use-templates-to-handle-response-data)
+with a list of weather services and their associated forecasts.
```
Comment (MartinHjelmare) [inline source/_integrations/weather.markdown]:
```suggestion
with a mapping of weather services and their associated forecasts.
```

File: source/_integrations/weather.markdown [hunk 14]
```diff
@@ -68,37 +67,29 @@ wind_gust_speed: 51.56
...
+    - weather.toronto_forecast
 data:
-  type: daily
+  type: hourly
 response_variable: weather_forecast
 ```
-{% endraw %}
 
-The response data field `forecast` is a list of forecasted conditions at a given point in time:
+The response data field is a list of called target entities, each containing the `forecast` field.
```
Comment (MartinHjelmare) [inline source/_integrations/weather.markdown]:
```suggestion
The response data field is a mapping of called target entities, each containing the `forecast` field.
```

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: source/_integrations/weather.markdown [hunk 15]
```diff
@@ -112,42 +103,60 @@ The response data field `forecast` is a list of forecasted conditions at a given
...
+| `temperature` | The temperature in the unit indicated by the `temperature_unit` state attribute. If `templow` is also provided, this is the higher temperature. | 14.2
 | `templow` | The lower temperature in the unit indicated by the `temperature_unit` state attribute. | 5.0
 | `uv_index` | The UV index. | 3
 | `wind_bearing` | The wind bearing in azimuth angle (degrees) or 1-3 letter cardinal direction. | 268
 | `wind_gust_speed` | The wind gust speed in the unit indicated by the `wind_speed_unit` state attribute. | 34.41
 | `wind_speed` | The wind speed in the unit indicated by the `wind_speed_unit` state attribute. | 24.41
 
 
-Example forecast:
+{% details "Example service response" %}
```
Comment (MartinHjelmare) [inline source/_integrations/weather.markdown:147]: I'd also add a simple example script or template sensor where the service is called and the response variable is accessed in a template. Not required for this PR though.

Comment (mikeodr) [reply source/_integrations/weather.markdown:147]: Takes just a minute and is a good addition.

File: source/_integrations/weather.markdown [hunk 16]
```diff
@@ -111,6 +111,39 @@ The response data field is a mapping of called target entities, each containing
...
+      - service: weather.get_forecasts
+        data:
+          type: hourly
+        target:
+          entity_id: weather.home
+        response_variable: hourly
+    sensor:
+      - name: Weather Forecast Hourly
+        unique_id: weather_forecast_hourly
+        state: "{{ now().isoformat() }}"
```
Comment (MartinHjelmare) [inline source/_integrations/weather.markdown]: I'd do something like this to not involve two different templates, and have less data in the example:

```yaml
sensor:
  - name: Temperature forecast next hour
    unique_id: temperature_forecast_next_hour
    state: "{{ hourly['weather.home'].forecast[0].temperature }}"
    unit_of_measurement: °C
```

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

**Pull request merged**